### ISSUE TITLE
[FEAT] update sampling mechanism to support iterators based dynamic loaders

### ISSUE THREAD

**Issue by Nadav-Barak** [deepchecks/deepchecks#2135]
We will stop working with complex sampler on top of pytorch. Will provide agnosticism that will allow easy integration to all batch iterators under the same mechanism 

In the future:  image paths will be provided via the new API
Now: image paths will not be provided, but everything will still work

Implement alternative shuffling mechanism (next slide)
Update the following check to not rely on the old mechanism: property outliers, label outliers, new labels, Robustness Report?

**Comment by Nadav-Barak** [comment]
Steps:
1. Reverify the suggested changes and finalize design - 2 SP
2. Change the relevant checks to not rely on old index based sampler - 3 SP
3. Update the sampler + test vision suites on iterator based DataLoader - 3 SP
4. Deprecate static predictions mechanism + update monitoring to not use it - 2 SP
5. Docs?

**Comment by Nadav-Barak** [comment]
Checks to modify: Image Property Outliers, Label Property Outliers, New Labels, 

Will not work it new API after step 1: Similar Image Leakage, Robustness Report
